hummingbot/connector/exchange/ndax/ndax_exchange.py

```diff
@@ -0,0 +1,134 @@
+                 trading_required: bool = True
+                 ):
+        """
+        :param uid: User ID of the account
+        :param api_key: The API key to connect to private NDAX APIs.
+        :param secret_key: The API secret.
+        :param trading_pairs: The market trading pairs which to track order book data.
+        :param trading_required: Whether actual trading is needed.
+        """
+        super().__init__()
+        self._trading_required = trading_required
+        self._trading_pairs = trading_pairs
+        self._auth = NdaxAuth(uid=uid, api_key=api_key, secret_key=secret_key)
+        # self._order_book_tracker = ProbitOrderBookTracker(trading_pairs=trading_pairs, domain=domain)
+        self._user_stream_tracker = NdaxUserStreamTracker(self._auth)
+        self._ev_loop = asyncio.get_event_loop()
+        self._shared_client = None
+        self._poll_notifier = asyncio.Event()
+        self._last_timestamp = 0
+        # self._in_flight_orders = {}  # Dict[client_order_id:str, ProbitInFlightOrder]
+        # self._order_not_found_records = {}  # Dict[client_order_id:str, count:int]
+        # self._trading_rules = {}  # Dict[trading_pair:str, TradingRule]
+        # self._last_poll_timestamp = 0
+
+        # self._status_polling_task = None
+        # self._user_stream_tracker_task = None
+        # self._user_stream_event_listener_task = None
+        # self._trading_rules_polling_task = None
+
+    @property
+    def name(self) -> str:
+        return CONSTANTS.EXCHANGE_NAME
+
+    def supported_order_types(self) -> List[OrderType]:
+        """
+        :return: a list of OrderType supported by this connector.
+        Note that Market order type is no longer required and will not be used.
+        """
+        return [OrderType.MARKET, OrderType.LIMIT, OrderType.LIMIT_MAKER]
+
+    async def _iter_user_event_queue(self) -> AsyncIterable[Dict[str, any]]:
+        while True:
+            try:
+                yield await self._user_stream_tracker.user_stream.get()
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(
+                    "Unknown error. Retrying after 1 seconds.",
+                    exc_info=True,
+                    app_warning_msg="Could not fetch user events from NDAX. Check API key and network connection."
+                )
+                await asyncio.sleep(1.0)
+
+    async def _user_stream_event_listener(self):
+        """
+        Listens to message in _user_stream_tracker.user_stream queue.
+        """
+        async for event_message in self._iter_user_event_queue():
+            try:
+                event_payload = NdaxMessagePayload.new_instance(
+                    endpoint=NdaxWebSocketAdaptor.endpoint_from_message(event_message),
+                    payload=NdaxWebSocketAdaptor.payload_from_message(event_message))
+                event_payload.process_event(connector=self)
```

Comment:
I like that you introduce the abstract class `NdaxMessagePayload`. But I have some reservations about this.
I think that it might be over abstracting things. 

The previous implementation simply parses and redirects the message to the appropriate processing methods of which we can clearly identify what it is doing. (i.e. `processing_trade_message()` or `processing_order_message()`). However with your implementation of `process_event()`, it is not immediately clear what is being done.